js_modules/dagster-ui/packages/ui-core/src/selection/SelectionInput.tsx

```diff
@@ -349,8 +344,26 @@ export const SelectionAutoCompleteInput = ({
             setShowResults({current: true});
           }}
         >
-          <Icon name="op_selector" style={{marginTop: 2}} />
+          <div style={{alignSelf: 'flex-start'}}>
+            <Icon name="op_selector" style={{marginTop: 2}} />
+          </div>
           <div ref={editorRef} />
+          {innerValue !== value ? (
+            <div
+              style={{alignSelf: 'flex-end'}}
+              ref={() => {
+                adjustHeight();
+              }}
+            >
+              <EnterHint>Enter</EnterHint>
+            </div>
+          ) : (
+            <div
+              ref={() => {
+                adjustHeight();
+              }}
+            />
+          )}
```

Comment:
- Why does it need the `adjustHeight()` in a ref?
- Would it help simplify things if the hint were absolutely positioned?

Reply:
the adjustHeight doesn't really need to be in a ref. I think I could do a `useLayoutEffect(adjustHeight ,[innerValue, value])`. I put in the ref so I wouldn't need to think about the layout effect condition but in this case its simple enough that I'll just do that. I was mostly just testing this pattern and I like it but overkill for this particular case

Reply:
Absolutely positioning wouldn't really help, Basically the input resizes so that the enter hint doesn't cover any text, and when the hint goes away it goes back to full width. Maybe instead I could permanently reserve the space for the enter hint but didn't really want to do that 😬

Reply:
Hmm, the nice thing about the ref approach is the colocation of the code. It's obvious that the div enter/leaving the DOM will affect the size of the codemirror input. with the layout effect its a step removed